### PR TITLE
MB-10110 Better Happo documentation

### PR DESCRIPTION
**Pull request by rogeruiz**
While @mlm55, @YanZ777, and I were troubleshooting transcom/mymove#7840, we discovered that Knobs don't seem to be supported by Happo on the CI/CD side. So this documentation was updated to help folks not run into the issues that came up during that pairing session.

[=> pairing on Happo](https://ustcdp3.slack.com/archives/C029R9LDTD4/p1638901935175200) 🔒 
[=> the fix that is being referenced in this `caution` admonition](https://github.com/transcom/mymove/pull/7840/commits/4a9fd6bab5a1dcb3143ac7f7c19ecef6e4ca798a)

### CODE REVIEW

File: docs/frontend/testing/frontend.md [hunk 1]
```diff
@@ -6,45 +6,6 @@ sidebar_position: 1
...
-  * [vi](#vi)
-  * [Browser Extensions](#browser-extensions)
-* [Learning](#learning)
-  * [JavaScript Concepts](#javascript-concepts)
-  * [Resources](#resources)
-
-Regenerate with "pre-commit run -a markdown-toc"
-
-<!-- tocstop -->
-
```
Comment (rogeruiz) [inline docs/frontend/testing/frontend.md:47]: Sorry for this deletion being included in this PR, but TOC don't need to be generated anymore because of Docusaurus. This page now includes a TOC on the right-hand side for Desktop.

Captura de Pantalla 2021-12-07 a la(s) 3 32 38 p m

File: docs/frontend/testing/frontend.md [hunk 2]
```diff
@@ -127,8 +86,29 @@ Historically we have leaned on Browser tests to cover testing our app thoroughly
...
+
+:::caution Storybook Addon Knobs
+The current version of Storybook used by Happo does not support [the library
+`@storybook/addon-knobs`][npm-storybook-addon-knobs] as expected. While it may
+work locally, Happo will not properly render the contents of components that are
+configured using these **Knobs**.
+
+> 👀 Because of this, make sure that you are passing actual JavaScript Object
+> de-structuring when setting your objects for Storybook components rather than
+> the `object()` function provided by the `@storybook/addon-knobs` library
```
Comment (rogeruiz) [inline docs/frontend/testing/frontend.md:107]: I'm looking for extra scrutiny here because I want to make sure that this is as clear as the commit that inspired this callout. https://github.com/transcom/mymove/pull/7840/commits/4a9fd6bab5a1dcb3143ac7f7c19ecef6e4ca798a

Comment (YanZ777) [reply docs/frontend/testing/frontend.md:107]: I'd rephrase this to: 
"pass in actual JavaScript Objects in your component props rather than using the `object()` function"

**Review by YanZ777 — COMMENTED** [review]
I'm going to post in the FE channel about this as well to check that what we're seeing is actually what is happening with Happo.

File: docs/frontend/testing/frontend.md (hunk 2 above)
Comment (rogeruiz) [reply docs/frontend/testing/frontend.md]: Captura de Pantalla 2021-12-08 a la(s) 11 41 35 a m

I've updated the admonition a bit to include a code example so it's easier to understand.

**Review by donaldthai — APPROVED** [review]
After reading the [slack thread](https://ustcdp3.slack.com/archives/CTQQJD3G8/p1638919185077400) if other folks were having issues with storybook knobs I think your changes look good.
